ark/schema/structure/optional.ts

```diff
@@ -95,23 +95,36 @@ export const Optional = {
 	Node: OptionalNode
 }
 
+const isPrimitive = (value: unknown): value is Primitive =>
+	typeof value === "object" ? value === null : typeof value !== "function"
+
 export const assertDefaultValueAssignability = (
 	node: BaseRoot,
 	value: unknown,
 	key = ""
 ): unknown => {
-	const out = node.in(value)
+	if (!isPrimitive(value) && typeof value !== "function")
```

Comment:
Let's just get rid of `isPrimitive` and instead use `hasDomain(value, "object")` (it works like the TS keyword so it handles this logic built in, and we use it throughout the repo)